static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,81 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+  clearProfile,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+
+class SettingsPage extends ProfileFormContainer {
+
+  componentDidMount() {
+    this.fetchProfile();
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile();
+  }
+
+  componentWillUnmount() {
+    const { dispatch, params: { username } } = this.props;
+    if (SETTINGS.username !== username) {
+      // don't erase the user's own profile from the state
+      dispatch(clearProfile(username));
+    }
+  }
+
+  startSettingsEdit() {
+    const { dispatch } = this.props;
+    const username = SETTINGS.username;
+    dispatch(startProfileEdit(username));
+  }
+
+  componentWillMount() {
+    this.startSettingsEdit();
+  }
+
+  render() {
+    const { profiles } = this.props;
+    const username = SETTINGS.username;
+    let profile;
+    let loaded = false;
+    let preferredName = SETTINGS.name;
+
+    if (profiles[username] !== undefined) {
+      profile = profiles[username];
+      loaded = profiles[username].getStatus !== FETCH_PROCESSING;
+      preferredName = profile.preferredName || SETTINGS.name;
+    }
+
+    let childrenWithProps = this.childrenWithProps(profile);
+
+    return (
+      <Loader loaded={loaded}>
+        <Jumbotron profile={ profile } text={ preferredName }>
+          <div className="card-copy">
+             { childrenWithProps }
+          </div>
+        </Jumbotron>
+      </Loader>
+    );
+  }
+}
+
+const mapStateToProps = state => {
+  let params = {
+    username: SETTINGS.username
+  };
+
+  return {
+    profiles: state.profiles,
+    ui: state.ui,
+    params: params
+  };
+};
+
+export default connect(mapStateToProps)(SettingsPage);
```

Comment:
Since we don't need to inject the `params` prop here you should be able to use `ProfileFormContainer.mapStateToProps` instead of declaring a new state -> props function here.